### PR TITLE
Add #612 Revisions

### PR DESCRIPTION
**Pull request by albinpa**

### CODE REVIEW

File: internal/migrate/migrate.go [hunk 1]
```diff
@@ -1024,3 +1027,47 @@ func migrateIdToStringInOrganisationCollection() {
 	}
 
 }
+
+func migrateSchemaNameAndAuthorizedByOtherInRevisionCollection() {
```
Comment (georgepadayatti) [inline internal/migrate/migrate.go:1031]: What does this migration function do ?

**Review by georgepadayatti — CHANGES_REQUESTED** [review]

**Comment by albinpa** [comment]
Instructions to run bdd test case ,

1.  Make deployable docker image from `main` branch of `bb-consent-api` by executing `make build/docker/deployable` command
2.  Copy `image name` and paste it in `image` fields value under `api` in `test-docker-compose.yaml` file inside test folder in bb-consent-playground
3.  Execute the following commands in `bb-consent-playground`
4.   `make destroy`
5.   `make build-fixtures`
6.   `make build-test`
7.   `make setup-test`
8.   `make run-test`
9.  Make deployable docker image from `feature/612` branch by executing `make build/docker/deployable` command in feature/612 branch
10. Copy image name and paste it in `image` fields value under `api` in `test-docker-compose.yaml` file inside `test` folder in bb-consent-playground
11.  Execute the following commands in `bb-consent-playground`
12.  `make destroy`
13.  `make setup-test`
14.  `make run-test`

**Comment by georgepadayatti** [comment]
I ran the tests from the `main` branch, and some failed as shown in the screenshot below. But the error is due to the 500 status code. This should not happen, instead, the response returned can contain an older field and fail due to that.

Can you check this ?

**Comment by albinpa** [comment]
> I ran the tests from the `main` branch, and some failed as shown in the screenshot below. But the error is due to the 500 status code. This should not happen, instead, the response returned can contain an older field and fail due to that.
> 
> Can you check this ?
> 
> 

The schemaName of data agreement in test data has value DataAgreement,  when trying to fetch revision using schemaName and objectId in delete data agreement in main branch which has schemaName as dataAgreement , the test fails.

**Comment by georgepadayatti** [comment]
Acceptance criteria for this PR to be merged: https://github.com/decentralised-dataexchange/bb-consent-playground/blob/main/gherkin/features/issue_612/issue_612.feature

Do implement the steps for the validating the above criteria against fix branch.

**Review by georgepadayatti — CHANGES_REQUESTED** [review]
Acceptance criteria for this PR to be merged: [link](https://github.com/decentralised-dataexchange/bb-consent-playground/blob/main/gherkin/features/issue_612/issue_612.feature)

Do implement the steps for the validating the above criteria against fix branch.

File: internal/revision/revisions.go [hunk 2]
```diff
@@ -21,24 +21,34 @@ type Revision struct {
...
 	ObjectData               string `json:"objectData"`
-	SuccessorId              string `json:"-"`
-	SerializedHash           string `json:"-"`
-	SerializedSnapshot       string `json:"-"`
+	SuccessorId              string `json:"SuccessorId"`
+	SerializedHash           string `json:"SerializedHash"`
+	SerializedSnapshot       string `json:"SerializedSnapshot"`
+}
+
+type RevisionForSnapshot struct {
```
Comment (georgepadayatti) [inline internal/revision/revisions.go:33]: Rename this to `SerialisedSnapshot` or `SerialisedSnapshotForDB`

**Review by georgepadayatti — CHANGES_REQUESTED** [review]

**Comment by georgepadayatti** [comment]
Acceptance criteria is met.

**Comment by georgepadayatti** [comment]
Acceptance criteria is met.

**Review by georgepadayatti — APPROVED** [review]
Acceptance criteria is met.

**Pull request merged**